File: spring-data-requery/src/test/java/org/springframework/data/requery/domain/sample/AbstractRole.java
package org.springframework.data.requery.domain.sample;

import io.requery.*;
import lombok.Getter;
import lombok.Setter;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.requery.domain.AbstractPersistable;
import org.springframework.data.requery.domain.ToStringBuilder;

import java.util.Objects;

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractRole
 *
 * @author debop
 * @since 18. 6. 14
 */
@Getter
@Setter
@Entity
@Table(name = "SD_Roles")
public abstract class AbstractRole extends AbstractPersistable<Integer> {

    private static final long serialVersionUID = -1819301866144058748L;

    public AbstractRole() {}

    public AbstractRole(String name) {
        this.name = name;
    }

    @Key
    @Generated
    protected Integer id;

    protected String name;


    @Override
    public int hashCode() {
        return Objects.hashCode(name);
    }

    @Transient
    @Override
    protected @NotNull ToStringBuilder buildStringHelper() {
        return super.buildStringHelper()
            .add("name", name);
    }
}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/domain/sample/AbstractRole.kt
package org.springframework.data.requery.domain.sample

import io.requery.Entity
import io.requery.Generated
import io.requery.Key
import io.requery.Table
import org.springframework.data.requery.domain.AbstractPersistable
import org.springframework.data.requery.domain.ToStringBuilder
import java.util.*

@Entity
@Table(name = "SD_Roles")
abstract class AbstractRole: AbstractPersistable<Int>() {

    @get:Key
    @get:Generated
    abstract override val id: Int

    abstract var name: String

    override fun hashCode(): Int {
        return Objects.hashCode(name)
    }

    override fun buildStringHelper(): ToStringBuilder {
        return super.buildStringHelper()
            .add("name", name)
    }
}